runtime/vm/ContinuationHelpers.cpp

```diff
@@ -247,30 +247,7 @@ enterContinuation(J9VMThread *currentThread, j9object_t continuationObject)
 		/* Continuation is still in a blocked list. This can happen with TIMED_WAIT.
 		 * It must be removed from the waiting list.
 		 */
-		bool foundInBlockedContinuationList = false;
-		bool foundInMonitorList = false;
-
-		omrthread_monitor_enter(vm->blockedVirtualThreadsMutex);
-
-		foundInBlockedContinuationList = VM_ContinuationHelpers::removeContinuationFromList(
-				&vm->blockedContinuations, continuation);
-
-		if (foundInBlockedContinuationList) {
-			continuation->objectWaitMonitor->virtualThreadWaitCount -= 1;
-			Assert_VM_true(continuation->objectWaitMonitor->virtualThreadWaitCount >= 0);
-		}
-
-		if (NULL != continuation->objectWaitMonitor->waitingContinuations) {
-			foundInMonitorList = VM_ContinuationHelpers::removeContinuationFromList(
-					&continuation->objectWaitMonitor->waitingContinuations, continuation);
-		}
-
-		omrthread_monitor_exit(vm->blockedVirtualThreadsMutex);
-
-		/* Virtual can only be in one list at a time. */
-		Assert_VM_false(foundInBlockedContinuationList && foundInMonitorList);
-
-		continuation->objectWaitMonitor = NULL;
+		Assert_VM_true(VM_ContinuationHelpers::removeBlockingContinuationFromLists(currentThread, continuation));
```

Comment:
This should not assume that `Assert_VM_true()` evaluates its argument.

Reply:
Just to confirm, it should be as follows if asserts are disabled?
```suggestion
		bool rc = VM_ContinuationHelpers::removeBlockingContinuationFromLists(currentThread, continuation)
		Assert_VM_true(rc);
```

Reply:
Yes, but with a better, more descriptive name than `rc`.